CONTRIBUTING.md

```diff
@@ -0,0 +1,94 @@
+
+If you want to make the changes yourself, or if you want to become a regular contributor, then you will want to set up [pre-commit](https://pre-commit.com/) on your local machine.  Once you do that, the CI checks will run locally before you even write your commit message.  This speeds up your development cycle considerably.
+
+## Getting Started
+
+### Contribution Process.
+
+Mentat follows the usual open source contribution process on Github. In order to contribute, you must fork the repository, create a branch based on the main branch, make your changes there and then create a pull request. Github has excellent documentation (and contributed documentation and videos) on how to do this. See [Contributing to a project](https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project), [first contributions](https://github.com/firstcontributions/first-contributions) for more details.
+
+The project, like most projects, does have some customizations specific to Mentat such as code formatting, quality checks which are enforced on each pull request using Github actions. Pull Requests (PRs - described later) are not accepted unless all the checks are passed.
+
+### Setup the Environment.
+
+Mentat requires Python (version 3.10 or later) for development. The source code comes with two sets of requirements. The first in requirements.txt are needed to run Mentat. The second additional requirements in dev-requirements.txt are needed to do development work. As a contributor you will need to install both sets of requirements to have the full set of tools. In addition you will need some tools outside of the python environment. 
+
+#### Python Virtual Environment setup
+
+For python it is best to setup a virtual environment dedicated to mentat development to avoid any conflicts with other projects. This section is not meant to be comprehensive in setting up python virtual environments. There are good tutorials on how to do that such as [Python Virtual Environments](https://www.arch.jhu.edu/python-virtual-environments/).  the notes below are supplemental to help with Mentat development.
+
+##### Conda (anaconda) Specific Notes
+
+Since not all the requirements are available in the conda repositories, it will be necessary to use pip to install the requirements. In order to avoid conflicts between installation using conda and pip you should add the following to your `.condarc` file (the location of the file varies based on the OS - but typically is in the users home directory):
+
+```
+pip_interop_enabled: true
+```
+
+## Commit Message Guidelines
+
+The guidelines below are for commit messages. While currently not enforced, eventually we will enforce the guidelines and use them for the [CHANGELOG.md][changelog-uri] generation and we strongly encourage following the guidelines.
```

Comment:
I like the idea of moving to this standard but maybe it would be good to loop in @biobootloader to discuss.

Reply:
I don't think we should include this section until we actually have commit message guidelines. Also, since we squash merge right now, it would be more important for these guidelines to apply to PR titles than commit messages.

Reply:
Made the changes